File: src/ast_build/key.c
#include "libft.h"

char	*get_key(char *key, char *tag, char *token, t_bool minus)
{
	if (minus)
	{
		if (ft_strequ(key, "") && ft_strequ(tag, "SPACES"))
		{
			ft_strdel(&key);
			return (ft_strdup(""));
		}
	}
	return (ft_freejoin(key, token));
}

int		quoted_gold_key(char *gold_key)
{
	char	last_char;

	last_char = gold_key[ft_strlen(gold_key) - 1];
	return (in(gold_key[0], "\"", "'", NULL) && \
			in(gold_key[len_key - 1], "\"", "'", NULL));
}

// Check if the key have to be allocated or if
//we can just perform change on the string.
void	modify_gold_key(char *gold_key)
{
	if (quoted_gold_key(gold_key))
	{
		gold_key[len_key - 1] = '\0';
		ft_strcpy(gold_key, gold_key + 1);
	}
}

File: pyshell/utils/key.py
#!/usr/bin/env python3

# import utils.global_var as gv


def get_key(key, tag, tok, minus):
    if minus:
        if key == '' and tag == 'SPACES':
            return ''
    return key + tok


def modifify_gold_key(gold_key):
    if gold_key[0] in '"\'' and gold_key[-1] in '"\'':
        return gold_key[1:-1]
    return gold_key


def quoted_gold_key(gold_key):
    return gold_key[0] in '"\'' and gold_key[-1] in '"\''


def dquoted_gold_key(gold_key):
    return gold_key[0] in '"' and gold_key[-1] in '"'
